algorithms/CPlusPlus/Graphs/cycle-detection.cpp

```diff
@@ -0,0 +1,89 @@
+// Program to detect whether a graph contains a cycle. The vertices of the graph are
+// placed in disjoint sets. As each edge is iterated through, the subset of the two
+// vertices is determined. If the subsets are the same, then a cycle is found.
+// Otherwise, the algorithm will join the two subsets together and repeat the process
+// until each edge is iterated through or a cycle is found.
+#include <algorithm>
+#include <iostream>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+// Edge list implementation of unweighted, undirected graph
+class Graph
+{
+    private:
+    std::vector<std::pair<int, int>> edgeList;
+    std::unordered_set<int> uniqueVertices;  // Keep track of the number of unique vertices
+    
+    public:                                                                                
+    void insertEdge(int from, int to);
+    bool isCycle();
+    int findSet(int parent[], int i);
+    int unionSet(int parent[], int x, int y);                                          
+};
+
+void Graph::insertEdge(int from, int to)
+{
+    edgeList.push_back(std::make_pair(from, to));
+    uniqueVertices.insert(from);
+    uniqueVertices.insert(to);
+}
+
+bool Graph::isCycle()
+{
+    // Initialize parent array as disjoint sets
+    int* parent = new int[uniqueVertices.size()];
+    std::fill(parent, parent + uniqueVertices.size(), -1);
```

Comment:
wouldn't it be easier to use vector for this? since you don't have to worry about manually de-allocating the memory, and saves you from an additional step.
```C
vector<int> parent(uniqueVertices.size(), -1);
```